packages/react/src/components/checkbox/checkbox.stories.tsx

```diff
@@ -66,6 +67,34 @@ export const AsIndeterminate = (props: CheckboxProps) => {
   return <Checkbox {...props} ref={checkboxRef} onChange={handleChange} />;
 };
 
+interface CheckboxesWithFieldsetLegendProps extends CheckboxProps {
+  name: string;
+  legend: string;
+}
+
+export const WithFieldsetLegend = ({
+  name,
+  legend,
+  ...restCheckboxProps
+}: CheckboxesWithFieldsetLegendProps) => (
+  <Fieldset>
+    <FieldsetLegend>{legend}</FieldsetLegend>
+    <Checkbox {...restCheckboxProps} name={name} value="1">
+      one
+    </Checkbox>
+    <Checkbox {...restCheckboxProps} name={name} value="2">
```

Comment:
💅🏻  There's no need to manually destructure the name here.

```suggestion
  legend,
  ...restCheckboxProps
}: CheckboxesWithFieldsetLegendProps) => (
  <Fieldset>
    <FieldsetLegend>{legend}</FieldsetLegend>
    <Checkbox {...restCheckboxProps} value="1">
      one
    </Checkbox>
    <Checkbox {...restCheckboxProps} value="2">
```